src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -240,6 +272,20 @@ export class WorkspaceSavedObjectsClientWrapper {
       id: string,
       options: SavedObjectsBaseOptions = {}
     ): Promise<SavedObject<T>> => {
+      if (this.isRelatedToWorkspace(type)) {
+        await this.validateSingleObjectPermissions(id, wrapperOptions.request, [
+          WorkspacePermissionMode.LibraryRead,
```

Comment:
Do we need to add `WorkspacePermissionMode.LibraryWrite` here?

Reply:
Do you think if there is a case that user can get saved object if he only has read permission?